docs/webhook.md

```diff
@@ -0,0 +1,57 @@
+# Webhook
+
+By default, Fleet utilizes polling (default: 15 seconds) to pull from a Git repo.However, this can be configured to utilize a webhook instead.Fleet currently supports Github,
+GitLab, Bitbucket, Bitbucket Server and Gogs.
+
+### 1. Configure the webhook service. Fleet uses `gitjob` service to handle webhook requests. Create an ingress that points to `gitjob` service.
+
+```yaml
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: webhook-ingress
+  namespace: fleet-system
+spec:
+  rules:
+  - host: your.domain.com
+    http:
+      paths:
+        - path: /
+          pathType: Prefix
+          backend:
+            serviceName: gitjob
+            servicePort: 80
+```
+
+!!! note
+    You can configure [TLS](https://kubernetes.io/docs/concepts/services-networking/ingress/#tls) on ingress. 
+
+### 2. Go to your webhook provider and configure the webhook callback url. Here is a Github example.
+
+![](./assets/webhook.png)
+
+Configuring secret is optional. This is used to validate webhook payload as payload should not be trusted by default. 
```

Comment:
Configuring secret is optional. This is used to validate the webhook payload as payload should not be trusted by default. 
If your webhook server is publicly accessible to the Internet, then it is recommended to configure the secret. If you do configure the 
secret, follow step 3.